### PR TITLE
[#21328] More Storybook stories

### PR DESCRIPTION
**Pull request by alfonsogoberjr**
## Description of the change

Adds the following components to Storybook:

- Modal
- MiniLoader
- DataViz/BarChart
- Toast

## Related issues

#21328

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [x] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by lilidworkin — COMMENTED** [review]
Very cool!! Will let @mxosman review/accept

File: common/components/DataViz/DatapointsView.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    disaggregationName: "Disaggregation",
+    countOrPercentageView: "Count",
+    setTimeRange: () => ({}),
+    setDisaggregationName: () => ({}),
+    setCountOrPercentageView: () => ({}),
+    metricName: "Funding",
+  },
+};
+
+// TODO: Needs more stories for optional props
```
[the remark this answers is not in the document]
Comment (mxosman) [reply common/components/DataViz/DatapointsView.stories.tsx:46]: I noticed it didn't like when I tried to set `datapointsGroupedByAggregateAndDisaggregations` (the display crashed with the error `Cannot read properties of undefined (reading 'Disaggregation')`) - wondering if that's related to your note about needing more stories for optional props?

**Review by mxosman — APPROVED** [review]
Ah - it's so wild to see these components alive in Storybook! Left various thoughts/comments throughout - but nothing really blocking or that can't be addressed in subsequent PRs. I imagine we'll continue iterating on this until everything is polished. Thank you so much for adding these complicated components.

File: common/components/Toast/Toast.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        timeout: 4000,
+      });
+      return (
+        <div id="toast" style={{ width: 200, height: 10, marginTop: 100 }}>
+          <Story />
+        </div>
+      );
+    },
+  ],
+};
```
[the remark this answers is not in the document]
Comment (alfonsogoberjr) [reply common/components/Toast/Toast.stories.tsx:88]: yeah this one is weird, I've been trying to fix it for a while

File: common/components/Modal/Modal.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      default: "gray",
+      values: [
+        {
+          name: "gray",
+          value: palette.highlight.grey1,
+        },
+      ],
+    },
+  },
+};
```
[the remark this answers is not in the document]
Comment (alfonsogoberjr) [reply common/components/Modal/Modal.stories.tsx:40]: hm let me check that

File: common/components/DataViz/BarChart.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        dataVizMissingData: 5,
+        Sentenced: 27098,
+        "Transfer or Hold": 41196,
+        Unknown: 28077,
+      },
+    ],
+    dimensionNames: ["Sentenced", "Transfoer or Hold", "Pretrial", "Unknown"],
+    percentageView: false,
+  },
+};
```
[the remark this answers is not in the document]
Comment (alfonsogoberjr) [reply common/components/DataViz/BarChart.stories.tsx:109]: It sounds more like an implementation thing, or test data not being correct. Let's de-scope that from this PR

File: common/components/DataViz/DatapointsView.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alfonsogoberjr) [reply common/components/DataViz/DatapointsView.stories.tsx:46]: Yes, so there are quite a few optional properties on this component, and this was just a placeholder to acknowledge that I haven't begun to really encapsulate all the permutations of state. Happy to remove it however

[the remark this answers is not in the document]
Comment (mxosman) [reply common/components/DataViz/DatapointsView.stories.tsx:46]: Ah - gotcha! Was mostly curious what the note meant - I don't mind it here as a reminder.

File: common/components/DataViz/BarChart.stories.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (lilidworkin) [reply common/components/DataViz/BarChart.stories.tsx:109]: @alfonsogoberjr @mxosman does it make sense to file a ticket to come back and look into this later?

[the remark this answers is not in the document]
Comment (mxosman) [reply common/components/DataViz/BarChart.stories.tsx:109]: I think so - none of this feels blocking! I think as we iterate, we can and should continue filing tickets on these stories until we fully flesh them out. Getting it to this point and rendering is a huge lift!

**Pull request merged**